subprojects/dependency-management/src/test/groovy/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/DefaultVersionedComponentChooserTest.groovy

```diff
@@ -252,30 +252,26 @@ class DefaultVersionedComponentChooserTest extends Specification {
             1 * selectedComponentResult.notMatched(d.id, _)
         } else {
             1 * selectedComponentResult.doesNotMatchConsumerAttributes({
-                it.id == d.id &&
-                    it.matchingDescription.find { it.requestedAttribute == Attribute.of('color', String) }
-                        .with { match ->
-                        assert match.requestedValue.get() == 'red'
-                        assert match.found.get() == 'blue'
-                        match
-                    }
+                it.id == d.id
+                assertMatchingDescription(it.matchingDescription, 'red', 'blue')
             })
         }
         1 * selectedComponentResult.doesNotMatchConsumerAttributes({
-            it.id == c.id &&
-                it.matchingDescription.find { it.requestedAttribute == Attribute.of('color', String) }
-                    .with { match ->
-                    assert match.requestedValue.get() == 'red'
-                    assert match.found.get() == 'green'
-                    match
-                }
+            it.id == c.id
```

Comment:
Is this a new Spock feature you are using here?

Reply:
yes, this behaves now as the assert blocks, evaluating each part separately: http://spockframework.org/spock/docs/1.3/release_notes.html#_code_argument_constraints_are_treated_as_implicit_assertions

If I change them they won't just show `false` but power assertion output is shown, see:
<img width="932" alt="Screen Shot 2019-06-05 at 17 32 38" src="https://user-images.githubusercontent.com/1841944/59085645-9f2bda00-88ff-11e9-9e37-7a5f614b835a.png">
<img width="932" alt="Screen Shot 2019-06-05 at 17 33 59" src="https://user-images.githubusercontent.com/1841944/59085646-9f2bda00-88ff-11e9-92e6-029adaf256d1.png">